front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/tag/TaggedEntity.groovy

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.front50.model.tag
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.netflix.spinnaker.front50.model.Timestamped
+
+class TaggedEntity implements Timestamped {
+  // {entityType}/{entityId}/{account}/{region}
+  String id
+
+  Long lastModified
+  String lastModifiedBy
+
+  Map<String, String> tags = [:]
+  EntityRef entityRef
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  static class EntityRef {
+    String entityType
+    String entityId
+
+    String cloudProvider
+    String application
+    String account
+    String region
```

Comment:
The list of tagged objects you gave as an example seems like the level of granularity I would expect.

And I like the idea of not having a fixed set of attributes in `EntityRef`.